snuba/clickhouse/query_profiler.py

```diff
@@ -0,0 +1,133 @@
+from datetime import datetime
+from typing import Optional, Sequence
+
+from snuba.clickhouse.query import Query
+from snuba.clickhouse.query_dsl.accessors import get_time_range
+from snuba.clickhouse.translators.snuba.mappers import (
+    KEY_COL_MAPPING_PARAM,
+    VALUE_COL_MAPPING_PARAM,
+    mapping_pattern,
+)
+from snuba.query.conditions import (
+    BooleanFunctions,
+    ConditionFunctions,
+    get_first_level_and_conditions,
+)
+from snuba.query.expressions import Column as ColumnExpr
+from snuba.query.expressions import Expression
+from snuba.query.expressions import FunctionCall as FunctionCallExpr
+from snuba.query.matchers import Any, Column, FunctionCall, Literal, Or, Param, String
+from snuba.querylog.query_metadata import (
+    ClickhouseQueryProfile,
+    Columnset,
+    FilterProfile,
+)
+
+
+def _get_date_range(query: Query) -> Optional[int]:
+    """
+    Best guess to find the time range for the query.
+    We pick the first column that is compared with a datetime Literal.
+    """
+    pattern = FunctionCall(
+        None,
+        Or([String(ConditionFunctions.GT), String(ConditionFunctions.GTE)]),
+        (Column(None, None, Param("col_name", Any(str))), Literal(None, Any(datetime))),
+    )
+
+    condition = query.get_condition_from_ast()
+    if condition is None:
+        return None
+    for exp in condition:
+        result = pattern.match(exp)
+        if result is not None:
+            from_date, to_date = get_time_range(query, result.string("col_name"))
+            if from_date is None or to_date is None:
+                return None
+            else:
+                return (to_date - from_date).days
+
+    return None
+
+
+def _get_table(query: Query) -> str:
+    source = query.get_data_source()
+    if source is None:
+        # Should never happen at this point.
+        return ""
+    return source.format_from()
+
+
+def _has_complex_conditions(query: Query) -> bool:
+    condition = query.get_condition_from_ast()
+    if condition is None:
+        return False
+    first_levels = get_first_level_and_conditions(condition)
```

Comment:
I don't think this will work the way you expect with Discover queries. To get around the schema limitations, OR queries are wrapped in `[x, "=", 1]` in a top level condition. In that case I believe the first level condition would be `equals(or(x, y), 1)` which wouldn't match.

Reply:
meh, ok, fixed, this though means we should revisit the implementation of get_first_level_and_conditions and get_first_level_or_conditions for other use cases.